### PR TITLE
Add section describing git hook functionality

### PR DESCRIPTION
**Pull request by caass**
This is a companion to [this PR](https://github.com/rust-lang/rust/pull/76356), which deals with including functionality for automatically running `tidy --bless` on each commit.

### CODE REVIEW

File: src/building/suggested.md [hunk 1]
```diff
@@ -58,7 +59,7 @@ Task] instead:
 }
 ```
 
-[Build Task]: https://code.visualstudio.com/docs/editor/tasks
+[build task]: https://code.visualstudio.com/docs/editor/tasks
```
Comment (jyn514) [inline src/building/suggested.md:62]: Did you mean to change this? I think this will break the link.

File: src/building/suggested.md [hunk 2]
```diff
@@ -3,6 +3,17 @@
...
+## Installing a `pre-commit` hook
+
+The `rust` CI will automatically fail your build if it doesn't pass `tidy`, our
+internal tool for ensuring code quality. If you'd like, you can install a
+[git hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
+that will automatically run `tidy --bless` on each commit, to ensure
+your code is up to par. If you decide later that this behavior is
+undesirable, you can delete the `pre-commit` file in `.git/hooks`.
+
+A prebuilt git hook lives at [`src/etc/pre-commit.sh`](https://github.com/rust-lang/rust/blob/master/src/etc/pre-commit.sh) which can be copied directly into your `.git/hooks` folder.
```
Comment (jyn514) [inline src/building/suggested.md:15]: It can't be copied directly, it has to be called `pre-commit`, not `pre-commit.sh`.

Comment (caass) [reply src/building/suggested.md:15]: ugh yeah thanks for reminding me

File: src/building/suggested.md (hunk 1 above)
Comment (caass) [reply src/building/suggested.md:62]: oh no i didn't realize this was changed, mb

File: src/building/suggested.md [hunk 3]
```diff
@@ -12,7 +12,9 @@ that will automatically run `tidy --bless` on each commit, to ensure
 your code is up to par. If you decide later that this behavior is
 undesirable, you can delete the `pre-commit` file in `.git/hooks`.
 
-A prebuilt git hook lives at [`src/etc/pre-commit.sh`](https://github.com/rust-lang/rust/blob/master/src/etc/pre-commit.sh) which can be copied directly into your `.git/hooks` folder.
+A prebuilt git hook lives at [`src/etc/pre-commit.sh`](https://github.com/rust-lang/rust/blob/master/src/etc/pre-commit.sh) which can be copied into your `.git/hooks` folder as `pre-commit` (without the `.sh` extension!).
+
+You can also install the hook as a step of running
```
Comment (jyn514) [inline src/building/suggested.md:17]: I think you a word.
```suggestion
You can also install the hook as a step of running `x.py setup`.
```

**Comment by camelid** [comment]
I believe this is no longer blocked, so I'm removing the label.

@rustbot modify labels: -blocked

File: src/building/suggested.md (hunk 3 above)
Comment (caass) [reply src/building/suggested.md]: argh

File: src/building/suggested.md [hunk 4]
```diff
@@ -3,6 +3,19 @@
 The full bootstrapping process takes quite a while. Here are some suggestions
 to make your life easier.
 
+## Installing a `pre-commit` hook
+
+The `rust` CI will automatically fail your build if it doesn't pass `tidy`, our
```
Comment (jyn514) [inline src/building/suggested.md:8]: tiny nit: I think this should either be `rust-lang/rust` or just 'Rust'.

**Review by jyn514 — APPROVED** [review]

File: src/building/suggested.md (hunk 4 above)
Comment (camelid) [reply src/building/suggested.md:8]: How about just "CI"? ;)

```suggestion
CI will automatically fail your build if it doesn't pass `tidy`, our
```

**Comment by camelid** [comment]
@jyn514 Nit is fixed!

@caass Could you [squash your commits](https://rustc-dev-guide.rust-lang.org/git.html?highlight=squash#advanced-rebasing)?

File: src/building/suggested.md [hunk 5]
```diff
@@ -3,6 +3,19 @@
 The full bootstrapping process takes quite a while. Here are some suggestions
 to make your life easier.
 
+## Installing a `pre-commit` hook
```
Comment (camelid) [inline src/building/suggested.md:6]: Actually, one more nit: I don't think it's necessary to have "pre-commit" as inline code.

```suggestion
## Installing a pre-commit hook
```

File: src/building/suggested.md [hunk 6]
```diff
@@ -3,6 +3,19 @@
...
+
+CI will automatically fail your build if it doesn't pass `tidy`, our
+internal tool for ensuring code quality. If you'd like, you can install a
+[git hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
```
Comment (camelid) [inline src/building/suggested.md:10]:
```suggestion
[Git hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
```

File: src/building/suggested.md [hunk 7]
```diff
@@ -3,6 +3,19 @@
...
+that will automatically run `tidy --bless` on each commit, to ensure
```
Comment (camelid) [inline src/building/suggested.md:11]:
```suggestion
that will automatically run `x.py test tidy --bless` on each commit, to ensure
```

**Review by camelid — COMMENTED** [review]
Oops, I have a few more nits. But they're not a big deal if you just want to get this merged :)

**Comment by caass** [comment]
@camelid no by all means! i'll fix them and pending anything else squash :)

**Pull request merged**